### PR TITLE
fix: Shared VPC architecture deployment and make Autoscaling optional

### PR DESCRIPTION
**Pull request by romancin**
### Changed
Depending on the plan you have on Terraform Cloud, the number of self hosted agents is limited: https://www.hashicorp.com/products/terraform/pricing

For Free and Standard plans, the maximum number of agents is only one, the autoscaling argument should be a variable so users can decide to enable autoscaling or not depending on the plan they use.

### Fixed
Currently, the module `tfc-agent-mig-container-vm` requires specifying network_name argument.

Following the Terraform documentation of compute_instance_template resource (https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_template#subnetwork_project), network, subnetwork and subnetwork_project are optional, but if we try to specify all of them, when the resource is created, it tries to find the network on the service project instead on the host project, so resource creation fails (although the plan seems to work fine). This is happening with the current release `0.1.0` of this module.

This release has a workaround to avoid this: if we set `subnet_name` variable, we force `network_name` to be null. This way, we can pass only `subnetwork_project`and `subnet_name` in case we use Shared VPC architecture, and the module will work fine. Also, we need to set a default value for `network_name` variable so Terraform plan don't fail.

### CODE REVIEW

**Comment by gautambaghel** [comment]
@romancin - Hi! This looks good to me, do you mind signing the CLA for us to merge the PR? thx!

**Comment by romancin** [comment]
> @romancin - Hi! This looks good to me, do you mind signing the CLA for us to merge the PR? thx!

Of course! We are going to sign it at company level. Reached my manager to do it, hope it is done tomorrow. Thank you!

**Comment by romancin** [comment]
> @romancin - can you please rebase this with master. Seems like you're forking this in an org account and I'll unable to do that for you.

Done!

**Comment by g-awmalik** [comment]
Sorry but its out of date again.

**Comment by romancin** [comment]
> Sorry but its out of date again.

@g-awmalik Rebased it again, no problem :D

**Comment by g-awmalik** [comment]
@romancin - can you take a look at the [lint errors](https://github.com/GoogleCloudPlatform/terraform-google-tf-cloud-agents/actions/runs/6590483386/job/17908931063?pr=10) please? You probably just need to run `make docker_generate_docs` and fix a minor formatting issue.

File: modules/tfc-agent-mig-container-vm/README.md [hunk 1]
```diff
@@ -23,12 +23,13 @@ This example shows how to deploy a self hosted Terraform Cloud agent on MIG Cont
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
 | additional\_metadata | Additional metadata to attach to the instance | `map(any)` | `{}` | no |
+| autoscaling_enabled | Set to true to enable autoscaling in the MIG | `bool` | `true` | no |
```
Comment (gautambaghel) [inline modules/tfc-agent-mig-container-vm/README.md:26]: `autoscaling\_enabled`

File: modules/tfc-agent-mig-container-vm/main.tf [hunk 2]
```diff
@@ -28,7 +28,7 @@ locals {
         path = "/var/run/docker.sock"
       }
   }] : []
-  network_name          = var.create_network ? google_compute_network.tfc_agent_network[0].self_link : var.network_name
+  network_name          = var.create_network ? google_compute_network.tfc_agent_network[0].self_link : ( var.subnet_name != "" ? null : var.network_name)
```
Comment (gautambaghel) [inline modules/tfc-agent-mig-container-vm/main.tf:31]: remove space before var here

**Review by gautambaghel — CHANGES_REQUESTED** [review]

File: modules/tfc-agent-mig-container-vm/main.tf (hunk 2 above)
Comment (romancin) [reply modules/tfc-agent-mig-container-vm/main.tf]: Done! Thank you!

**Review by gautambaghel — APPROVED** [review]

File: modules/tfc-agent-mig-container-vm/README.md [hunk 3]
```diff
@@ -70,4 +71,4 @@ Before this module can be used on a project, you must ensure that the following
     "storage-component.googleapis.com",
     "logging.googleapis.com",
     "monitoring.googleapis.com"
-    ```
+    ```
```
Comment (gautambaghel) [inline modules/tfc-agent-mig-container-vm/README.md:74]: @romancin can you please add a newline at the end of this file, silly lint error

Comment (romancin) [reply modules/tfc-agent-mig-container-vm/README.md]: Done @gautambaghel !

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**